app/models/service_template_transformation_plan_task.rb

```diff
@@ -81,6 +82,20 @@ def preflight_check
     { :status => 'Error', :message => error.message }
   end
 
+  def preflight_check_vm_exists_in_destination
+    send("preflight_check_vm_exists_in_destination_#{destination_ems.emstype}")
+  end
+
+  def preflight_check_vm_exists_in_destination_rhevm
+    vms = Vm.where(:name => source.name, :ems_cluster => destination_cluster)
+    raise "A VM named '#{source.name}' already exist in destination provider" unless vms.empty?
+  end
+
+  def preflight_check_vm_exists_in_destination_openstack
+    vms = Vm.where(:name => source.name, :cloud_tenant_id => destination_cluster.id)
```

Comment:
Same comment as above but also any reason why you're using the `_id` here instead of just `:cloud_tenant => destination_cluster` ?

Reply:
When I try to use it, I have a database error. Probably something with the relationship definition. I was a bit lazy here and use `cloud_tenant_id`:

```
irb> my_tenant = CloudTenant.find_by(name: 'my_tenant')
irb> my_vms = Vm.where(:name => 'my_vm', :cloud_tenant => my_tenant)
Traceback (most recent call last):
       16: from activerecord (5.1.7) lib/active_record/connection_adapters/abstract/database_statements.rb:42:in `select_all'
       15: from activerecord (5.1.7) lib/active_record/connection_adapters/abstract/database_statements.rb:371:in `select'
       14: from activerecord (5.1.7) lib/active_record/connection_adapters/postgresql/database_statements.rb:79:in `exec_query'
       13: from activerecord (5.1.7) lib/active_record/connection_adapters/postgresql_adapter.rb:611:in `execute_and_clear'
       12: from activerecord (5.1.7) lib/active_record/connection_adapters/postgresql_adapter.rb:622:in `exec_no_cache'
       11: from activerecord (5.1.7) lib/active_record/connection_adapters/abstract_adapter.rb:604:in `log'
       10: from activesupport (5.1.7) lib/active_support/notifications/instrumenter.rb:21:in `instrument'
        9: from activerecord (5.1.7) lib/active_record/connection_adapters/abstract_adapter.rb:612:in `block in log'
        8: from /usr/share/ruby/monitor.rb:226:in `mon_synchronize'
        7: from activerecord (5.1.7) lib/active_record/connection_adapters/abstract_adapter.rb:613:in `block (2 levels) in log'
        6: from activerecord (5.1.7) lib/active_record/connection_adapters/postgresql_adapter.rb:623:in `block in exec_no_cache'
        5: from activesupport (5.1.7) lib/active_support/dependencies/interlock.rb:45:in `permit_concurrent_loads'
        4: from activesupport (5.1.7) lib/active_support/concurrency/share_lock.rb:185:in `yield_shares'
        3: from activesupport (5.1.7) lib/active_support/dependencies/interlock.rb:46:in `block in permit_concurrent_loads'
        2: from activerecord (5.1.7) lib/active_record/connection_adapters/postgresql_adapter.rb:624:in `block (2 levels) in exec_no_cache'
        1: from activerecord (5.1.7) lib/active_record/connection_adapters/postgresql_adapter.rb:624:in `exec_params'
ActiveRecord::StatementInvalid (PG::UndefinedColumn: ERROR:  column vms.cloud_tenant does not exist)
LINE 1: ...D "vms"."template" = $1 AND "vms"."name" = $2 AND "vms"."clo...
                                                             ^
HINT:  Perhaps you meant to reference the column "vms.cloud_tenant_id".
: SELECT  "vms".* FROM "vms" WHERE "vms"."type" IN ('Vm', 'ManageIQ::Providers::InfraManager::Vm', 'ManageIQ::Providers::CloudManager::Vm', 'VmServer', 'ManageIQ::Providers::Vmware::InfraManager::Vm', 'ManageIQ::Providers::Kubevirt::InfraManager::Vm', 'ManageIQ::Providers::Redhat::InfraManager::Vm', 'ManageIQ::Providers::Microsoft::InfraManager::Vm', 'VmXen', 'ManageIQ::Providers::Amazon::CloudManager::Vm', 'ManageIQ::Providers::Azure::CloudManager::Vm', 'ManageIQ::Providers::AzureStack::CloudManager::Vm', 'ManageIQ::Providers::Google::CloudManager::Vm', 'ManageIQ::Providers::Openstack::CloudManager::Vm', 'ManageIQ::Providers::Vmware::CloudManager::Vm') AND "vms"."template" = $1 AND "vms"."name" = $2 AND "vms"."cloud_tenant" = 9 LIMIT $3
```

Reply:
Opened https://bugzilla.redhat.com/show_bug.cgi?id=1809580 to track it.
Proposed fix: https://github.com/ManageIQ/manageiq/pull/19914.